### PR TITLE
Add Trust

### PR DESCRIPTION
**Pull request by Kizr**
## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated or [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/services/gias_csv_importer.rb [hunk 1]
```diff
@@ -87,54 +88,68 @@ def initialize(csv_path)
 
   def call
     invalid_records = []
-    records = []
+    school_records = []
+    trust_records = []
+    trust_associations = Hash.new { |h, k| h[k] = [] }
```
Comment (ollietreend) [inline app/services/gias_csv_importer.rb:93]: This is clever! 🤓 I've never seen this before.

File: app/services/gias_csv_importer.rb [hunk 2]
```diff
@@ -171,6 +186,17 @@ def determine_region(district_admin_code)
...
   end
 
+  def associate_schools_to_trusts(trust_data)
+    Rails.logger.debug "Associating schools to trusts... "
+
+    trust_data.each do |uid, urns|
+      trust = Trust.find_by(uid:)
+      next unless trust
+
+      School.where(urn: urns).update_all(trust_id: trust.id)
```
Comment (ollietreend) [inline app/services/gias_csv_importer.rb:196]: It's only syntactic sugar, but I think you can simplify this to:

```suggestion
      School.where(urn: urns).update_all(trust:)
```

**Review by ollietreend — COMMENTED** [review]
Just a couple of little things from me – but I'm not approving because this PR is still in draft.

File: app/services/gias_csv_importer.rb (hunk 2 above)
Comment (JamieCleare2525) [reply app/services/gias_csv_importer.rb:196]: I don't think `update_all` accepts this.
As I replied in https://github.com/DFE-Digital/itt-mentor-services/pull/345#discussion_r1513041291, `update_all(region: inner_london)` returns a `TypeError: can't cast Region (TypeError)`

Comment (Kizr) [reply app/services/gias_csv_importer.rb:196]: Yeah, update_all needs the `id` here, but if I could use the nicer syntax I would!

File: app/services/gias_csv_importer.rb (hunk 1 above)
Comment (Kizr) [reply app/services/gias_csv_importer.rb:93]: I've only learnt it myself recently, I think it's an improvement over adding a guard clause to create the empty array should it be missing.

File: spec/models/trust_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#
+require "rails_helper"
+
+RSpec.describe Trust, type: :model do
+  describe "associations" do
+    it { is_expected.to have_many(:schools) }
+  end
+
+  describe "validations" do
+    subject { create(:trust) }
```
[the remark this answers is not in the document]
Comment (Kizr) [reply spec/models/trust_spec.rb]: Great spot, thank you!

File: app/services/gias_csv_importer.rb (hunk 2 above)
Comment (ollietreend) [reply app/services/gias_csv_importer.rb:196]: Oops. I missed @JamieCleare2525's reply on the other PR comment 🙃

**Pull request closed without merging**